### ISSUE TITLE
Resource update should be resource independent

### ISSUE THREAD

**Issue by Zharktas** [ckan/ckan#4995]
### CKAN Version if known (or site URL)
2.8, probably master and every other ckan version

### Please describe the expected behaviour
Calling resource update on a resource should only update and validate the single resource.

### Please describe the actual behaviour
Resource update calls package update which validates all the resources which might fail if the resources have wrong values in them.

### What steps can be taken to reproduce the issue? 
Create dataset.
Create two resources in to dataset.
Make one the empty resource fields required in the datamodel.
Call resource update on one resource with required field filled.
Validation will fail as the other resource does not have the required field.

**Comment by wardi** [comment]
I think this change will just push the problem further down. Having a required field missing could cause a failure when indexing the dataset, or even just when displaying the resource.

If you change the data model I think it's up to you to migrate your data to match.

Another option is to add a field at the dataset or resource level that indicates a schema version then the required-resource-field validation can be dependent on that field.

If we split up the validation to be per-resource then it might be possible to manually migrate records one at a time using the resource form, but we lose the ability to have fields at the dataset level affect resource validation (other than the dataset `type` which selects the whole schema).

**Comment by boykoc** [comment]
@Zharktas do you know if this effects the `last_updated` or is this just checking for validations on all the resources?  If it's calling package_update, I'm curious if it's actually triggering updates on all resources instead of just the one. I'm thinking of issue #4951

**Comment by Zharktas** [comment]
@boykoc As far as I know, package_update only validates resources, it does not update them in any way.

**Comment by wardi** [comment]
@Zharktas package_update can be used to update all package and resource metadata (that's how `ckanapi load datasets` works) The only thing package_update can't do right now is upload resource files themselves. I've enabled uploading resource files as part of package_update in #4618, which I hope to be merged soon.

**Comment by Zharktas** [comment]
@wardi yeah, we ended up using package_update, it's just counter intuitive that you just can't update all the resources individually if the have validation errors. :)
